# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../lzma/compile.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -ur lzma-4.27.1/sdk/SRC/7zip/Compress/LZMA_Alone/Makefile lzma-4.27.1-fixed/sdk/SRC/7zip/Compress/LZMA_Alone/Makefile
--- lzma-4.27.1/sdk/SRC/7zip/Compress/LZMA_Alone/Makefile	2005-08-24 19:15:58.000000000 +0200
+++ lzma-4.27.1-fixed/sdk/SRC/7zip/Compress/LZMA_Alone/Makefile	2005-10-02 21:33:47.000000000 +0200
@@ -4,7 +4,8 @@
 PROG = lzma
 
 ## Compile only bt4, hc3 and hc4 match finders:
-CXX = gcc -DCOMPRESS_MF_BT -DCOMPRESS_MF_BT4 -DCOMPRESS_MF_HC \
+CXX = gcc
+override CXX +=  -DCOMPRESS_MF_BT -DCOMPRESS_MF_BT4 -DCOMPRESS_MF_HC \
 		-I ../../../ -c $(CFLAGS)
 LD = gcc $(CFLAGS)
 
diff -ur lzma-4.27.1/utils/Makefile lzma-4.27.1-fixed/utils/Makefile
--- lzma-4.27.1/utils/Makefile	2005-08-24 17:56:46.000000000 +0200
+++ lzma-4.27.1-fixed/utils/Makefile	2005-10-02 21:32:42.000000000 +0200
@@ -1,8 +1,9 @@
-## -D_LZMA_IN_CB and -D_LZMA_OUT_READ are required by lzmadec.
+
 ## -D_LZMA_SYSTEM_SIZE_T should keep it working also on 64-bit arch.
 ## -D_LZMA_PROB32 is a speed optimization.
 
-CC = gcc -D_LZMA_IN_CB -D_LZMA_OUT_READ -D_LZMA_SYSTEM_SIZE_T -D_LZMA_PROB32
+CC = gcc
+override CC += -D_LZMA_IN_CB -D_LZMA_OUT_READ -D_LZMA_SYSTEM_SIZE_T -D_LZMA_PROB32
 LD = gcc
 LDFLAGS = $(CFLAGS)
 RM = rm -f
--- lzma-4.27.1/Makefile.vanilla	2005-10-02 21:49:17.000000000 +0200
+++ lzma-4.27.1/Makefile	2005-10-02 21:50:47.000000000 +0200
@@ -20,16 +20,16 @@
 	cd sdk/SRC/7zip/Compress/LZMA_Alone && $(MAKE)
 
 install: all
-	$(INSTALL) -dm 0755 $(bindir) $(mandir)/man1 \
-			$(docdir)/$(docdir_name)
-	$(INSTALL) -pm 0755 utils/lzmadec utils/lzmainfo \
+#	$(INSTALL) -dm 0755 $(bindir) $(mandir)/man1 \
+#			$(docdir)/$(docdir_name)
+	$(INSTALL) -m 0755 utils/lzmadec utils/lzmainfo \
 			utils/lzmash utils/lzdiff utils/lzmore utils/lzgrep \
 			sdk/SRC/7zip/Compress/LZMA_Alone/lzma \
 			$(bindir)
-	$(INSTALL) -pm 0644 man/lzma.1 man/lzmadec.1 man/lzmainfo.1 \
+	$(INSTALL) -m 0644 man/lzma.1 man/lzmadec.1 man/lzmainfo.1 \
 			man/lzmash.1 man/lzdiff.1 man/lzgrep.1 man/lzmore.1 \
 			$(mandir)/man1
-	$(INSTALL) -pm 0644 BENCHMARKS COPYING.GPL COPYING.LGPL INSTALL \
+	$(INSTALL) -m 0644 BENCHMARKS COPYING.GPL COPYING.LGPL INSTALL \
 			README README.LICENSES THANKS TODO sdk/lzma.txt \
 			$(docdir)/$(docdir_name)
 	$(LN) lzmash $(bindir)/lunzmash
--- lzma-4.27.1/Makefile.vanilla	2005-10-02 21:53:59.000000000 +0200
+++ lzma-4.27.1/Makefile	2005-10-02 21:54:24.000000000 +0200
@@ -8,8 +8,10 @@
 docdir_name = lzma-4.27.1
 
 INSTALL = install
-RM = rm -f
-LN = ln -s
+RM = rm
+override RM += -f
+LN = ln
+override LN += -sf
 
 all: utils sdk
 
